lib/cfn-model/parser/lambda_function_parser.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+require 'cfn-model/model/lambda_function'
+require 'cfn-model/model/references'
+
+# Compute the role_id for parsed Lambda function
+class LambdaFunctionParser
+  def parse(cfn_model:, resource:)
+    lambda_function = resource
+
+    wire_role_to_lambda_function(cfn_model, lambda_function)
+
+    lambda_function
+  end
+
+  private
+
+  def wire_role_to_lambda_function(cfn_model, lambda_function)
+    if lambda_function.role.instance_of?(String)
+      lambda_function.role_object = nil
+      return
+    end
+
+    role_id = References.resolve_resource_id(lambda_function.role)
```

Comment:
this is the logical_resource_id?  when i read this code - i was sure we had a method to do this in CfnModel but apparently we don't!  lol so .... i would suggest generalizing this piece of code to find the role_object and push it up to CfnModel

Reply:
In the latest commit I created a `resource_by_ref` function in CfnModel, which will accept a reference from an object, determine the correct logical_resource_id, then return a matching resource from that ID.